Validation/MuonGEMRecHits/plugins/GEMRecHitValidation.cc

```diff
@@ -12,12 +12,13 @@ GEMRecHitValidation::GEMRecHitValidation(const edm::ParameterSet& pset)
   const auto& simhit_pset = pset.getParameterSet("gemSimHit");
   const auto& simhit_tag = simhit_pset.getParameter<edm::InputTag>("inputTag");
   simhit_token_ = consumes<edm::PSimHitContainer>(simhit_tag);
+  geomToken_ = esConsumes<GEMGeometry, MuonGeometryRecord>();
```

Comment:
You will need two different tokens, one for getting the data during the BeginRun and the other for getting it during the Event. When doing the esConsumes call for the BeginRun token you will use a third template argument edm::Transition::BeginRun